### PR TITLE
test(accessibility-checker): update to `versioned` ruleArchive

### LINKED ISSUE #17149: Update to `versioned` accessibility-checker config, latest ruleset
In https://github.com/carbon-design-system/carbon/pull/17128 accessibility-checker was updated. We can now make use of the `versioned` ruleset.

`versioned` makes sure we use a rule pack that matches the version of accessibility-checker we're currently on. New rulesets will only be brought in by updating accessibility-checker, so no more randomly failing CI checks out of the blue after a new ruleset is published on latest. 

This will effectively update the ruleset to the latest, `06Aug2024`. There will be violations/failures we'll need to sort through and fix and/or turn off certain rules that do not apply to storybook or the context in which we're developing the components.

```[tasklist]
### Potential related violations
- [ ] https://github.com/carbon-design-system/carbon/issues/17176
- [ ] https://github.com/carbon-design-system/carbon/issues/17223
- [ ] From #16582: 1. DatePicker @avt-advanced-states open test skipped due to color contrast a11y failure and 2. Skipped DataTable's @avt-advanced-states with-radio-expansion test due to an odd failure where the id's on rows are not correct on the dynamic story
- [ ] https://github.com/carbon-design-system/carbon/issues/18202
```

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #17149 

#### Changelog

**Changed**

- Update accessibility-checker config to use `versioned` ruleArchive (effectively ~`06Aug2024`~ ~`29Aug2024`~ ~`01Oct2024`~ `13Nov24` as of this PR)

- Adds  `aria_attribute_conflict`, `label_name_visible` and `target_spacing_sufficient` rules to denylist inside toHaveNoACViolations of playwright and jest configs to suppress false positive violations.
- These should be removed after this PR gets merged -> [Issue](https://github.com/carbon-design-system/carbon/issues/18227)
- Added test cases for `SliderSkeleton`, moved `Slider-test.js `and `SliderSkeleton-test.js` to ` __test__ ` folder

### CODE REVIEW

**Comment by tay1orjones** [comment]
I've updated this PR to the latest version of `accessibility-checker`, which puts the `versioned` rule archive at the new Aug29 one.

Current failures:
```
  8 failed
    [chromium] › components/AILabel/Slug-test.avt.e2e.js:14:3 › @avt AILabel › @avt-default-state ──
    [chromium] › components/ComboBox/ComboBox-test.avt.e2e.js:14:3 › @avt ComboBox › @avt-default-state
    [chromium] › components/FluidComboBox/FluidComboBox-test.avt.e2e.js:14:3 › @avt FluidComboBox › @avt-default-state
    [chromium] › components/Menu/Menu-test.avt.e2e.js:25:3 › @avt Menu › @avt-keyboard-nav Menu ────
    [chromium] › components/MultiSelect/MultiSelect-test.avt.e2e.js:25:3 › @avt MultiSelect › @avt-advanced-states filterable multiselect
    [chromium] › components/Slider/Slider-test.avt.e2e.js:51:3 › @avt Slider › @avt-advanced-states two handle slider
    [chromium] › components/Slider/Slider-test.avt.e2e.js:73:3 › @avt Slider › @avt-advanced-states two handle skeleton
    [chromium] › components/Toggle/Toggle-test.avt.e2e.js:36:3 › @avt Toggle › @avt-advanced-states small toggle
```

**Comment by tay1orjones** [comment]
This has now been updated again to the latest `Oct 1` ruleset

File: config/jest-config-carbon/matchers/toHaveNoACViolations.js [hunk 1]
```diff
@@ -20,6 +20,9 @@ async function toHaveNoACViolations(node, label) {
       'html_skipnav_exists',
       'aria_content_in_landmark',
       'aria_child_tabbable',
+      'aria_attribute_conflict',
+      'label_name_visible',
+      'target_spacing_sufficient',
```
Comment (tay1orjones) [inline config/jest-config-carbon/matchers/toHaveNoACViolations.js:25]: Do we have issues made to track getting these rules re-enabled after resolving the false positives?

**Review by ariellalgilmore — APPROVED** [review]
LGTM!

File: packages/styles/scss/components/text-area/_text-area.scss [hunk 2]
```diff
@@ -29,7 +29,7 @@
     @include type-style('body-01');
     @include focus-outline('reset');
 
-    padding: convert.to-rem(11px) layout.density('padding-inline');
+    padding: convert.to-rem(12px) layout.density('padding-inline');
```
Comment (alisonjoseph) [inline packages/styles/scss/components/text-area/_text-area.scss]: Just double checking this update was intentional?

**Review by alisonjoseph — APPROVED** [review]
This all LGTM! One small question and did you want to add any tests for the added props in Slider.Skeleton before we merge this?

File: packages/styles/scss/components/text-area/_text-area.scss (hunk 2 above)
Comment (2nikhiltom) [reply packages/styles/scss/components/text-area/_text-area.scss]: Looks unintentional. @tay1orjones : I reverted this change, but just wanted to check if there was a specific reason for adding it.

Comment (tay1orjones) [reply packages/styles/scss/components/text-area/_text-area.scss]: Yeah this was originally failing the minimum target size rule and the 1px seemed to be a defect when I looked more closely at it. We can remove if it passes without it 👍

**Pull request merged**